.circleci/config.yml

```diff
@@ -0,0 +1,198 @@
+linuxkit_pkg_build: &linuxkit_pkg_build
+  docker:
+    - image: debian:stretch
+  steps:
+    - run:
+        name: Configure $PATH
+        command: echo 'export PATH=/workspace/bin:$PATH' >> $BASH_ENV
+    - run:
+        name: Install packages
+        # ca-certificates are needed for attach_workspace (and git over https)
+        command: apt-get update && apt-get install -y ca-certificates git openssh-client
+    - attach_workspace:
+        at: /workspace
+    - checkout
+    - setup_remote_docker:
+        version: 17.06.1-ce
+    - run:
+        name: Docker version
+        command: |
+          docker version
+    - run:
+        name: Building package
+        command: |
+          PKG=${CIRCLE_JOB#pkg-}
+          mkdir /workspace/images
+          linuxkit pkg build pkg/$PKG
+          docker image ls
+          linuxkit pkg show-tag pkg/$PKG > /workspace/images/$PKG.tag
+          docker save -o /workspace/images/$PKG.tar $(cat /workspace/images/$PKG.tag)
+    - persist_to_workspace:
+        root: /workspace
+        paths: images
+
+image_cache_build: &image_cache_build
+  docker:
+    - image: debian:stretch
+  steps:
+    - run:
+        name: Configure $PATH
+        command: echo 'export PATH=/workspace/bin:$PATH' >> $BASH_ENV
+    - run:
+        name: Install packages
+        # ca-certificates are needed for attach_workspace (and git over https)
+        command: apt-get update && apt-get install -y ca-certificates git make openssh-client
+    - attach_workspace:
+        at: /workspace
+    - checkout
+    - setup_remote_docker:
+        version: 17.06.1-ce
+    - run:
+        name: Docker version
+        command: |
+          docker version
+    - run:
+        name: Building package
+        command: |
+          CACHE=${CIRCLE_JOB#pkg-image-cache-}
+          PKG=kubernetes-docker-image-cache-$CACHE
+          mkdir /workspace/images
+          make --no-print-directory -C pkg/image-cache build-$CACHE
+          make --no-print-directory -C pkg/image-cache show-tag-$CACHE > /workspace/images/$PKG.tag
+          docker save -o /workspace/images/$PKG.tar $(cat /workspace/images/$PKG.tag)
+    - persist_to_workspace:
+        root: /workspace
+        paths: images
+
+version: 2
+jobs:
+  dependencies:
+    docker:
+      - image: debian:stretch
+    steps:
+      - run:
+          name: Create workspace
+          command: mkdir -p /workspace/bin
+      - run:
+          name: Install packages
+          command: apt-get update && apt-get install -y ca-certificates curl
+      - run:
+          name: Fetch binaries
+          command: |
+            curl -fsSL -o /workspace/bin/linuxkit https://152-46932243-gh.circle-artifacts.com/0/linuxkit-linux-amd64
+            curl -fsSL -o /workspace/bin/moby https://309-89286561-gh.circle-artifacts.com/0/moby-linux
+            curl -fsSL -o /tmp/docker.tgz https://download.docker.com/linux/static/stable/x86_64/docker-17.06.2-ce.tgz
```

Comment:
Is there a reason you are downloading `docker` here while above you only say:
```
- setup_remote_docker:
     version: 17.06.1-ce
```

Reply:
Download is the client, the `setup_remote_docker` is the server. 17.06.1 is the latest point of 17.06 circle provides for the server side which is why the client is a bit newer.